### PR TITLE
Feature/300 role based nav

### PR DESCRIPTION
**Pull request by mabiede**
- add role-based navigation bar (extract layout to own module)
- add cached version of guardian validation
- typify icon component
- update icon references
- use message from layout
- remove layout and message in page module

### CODE REVIEW

**Review by timohuber — CHANGES_REQUESTED** [review]
@mabiede As discussed, please have a look at the main navigation as `LocationManager`.

File: pool/routes/routes.ml [hunk 1]
```diff
@@ -385,7 +385,7 @@ module Admin = struct
       let assignments =
         let open Experiments.Assignment in
         [ get "" ~middlewares:[ Access.index ] index
-        ; get "deleted" ~middlewares:[ Access.deleted ] deleted
+        ; get "deleted" ~middlewares:[ Access.delete ] deleted
```
Comment (timohuber) [inline pool/routes/routes.ml:388]: This middleware does not work. The route `admin/experiments/983a1f96-5862-4b86-91fa-0f77d2e02ed7/assignments/deleted` does not contain an assignment_id. It is a get request to show all deleted assignments.

IMO we can use the `index` access.

File: pool/routes/routes.ml [hunk 2]
```diff
@@ -555,10 +555,7 @@ module Admin = struct
       in
       let models =
         [ get "" ~middlewares:[ Access.index ] index
-        ; post
-            "sort-fields"
-            ~middlewares:[ Access.sort_ungrouped ]
-            sort_ungrouped_fields
+        ; post "sort-fields" ~middlewares:[ Access.sort ] sort_ungrouped_fields
```
Comment (timohuber) [inline pool/routes/routes.ml:558]: This middleware does not work (also sort groups). They both throw `Not_found` exception. They both look for an id in the route.

File: pool/schedule/entity_guard.ml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `Schedule)
+  let read = One (Action.Read, TargetSpec.Entity `Schedule)
```
Comment (timohuber) [inline pool/schedule/entity_guard.ml:6]: Only one of those is required

File: pool/web/view/layout/entity.ml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+module I18n = Pool_common.I18n
+module Icon = Component.Icon
+module Language = Pool_common.Language
+
+module Element = struct
```
Comment (timohuber) [inline pool/web/view/layout/entity.ml:5]: Please use another name, e.g. `NavLink`/`NavElement`

**Review by timohuber — CHANGES_REQUESTED** [review]

File: pool/app/contact/entity_guard.ml [hunk 5]
```diff
@@ -1,14 +1,59 @@
...
+  ;;
+
+  let to_authorizable ?ctx =
+    let encode = Entity.id %> Uuid.actor_of Pool_common.Id.value in
+    decorate ?ctx encode
+  ;;
+
+  (** Many request handlers do not extract a [User.t] at any point. This
+      function is useful in such cases. *)
+  let authorizable_of_req ?ctx req =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/contact/entity_guard.ml:40]: I guess is a leftover from rewrite version, is removed.

File: pool/app/custom_field/entity_guard.ml [hunk 6]
```diff
@@ -18,13 +23,35 @@ module Group = struct
...
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `CustomField)
+
+  let read id =
+    let target_id = id |> Uuid.target_of Entity.Id.value in
+    One (Action.Read, TargetSpec.Id (`CustomField, target_id))
+  ;;
+
+  module Group = struct
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/custom_field/entity_guard.ml]: They are now used by the group commands.

File: pool/app/custom_field/entity_guard.ml [hunk 7]
```diff
@@ -18,13 +23,35 @@ module Group = struct
...
+  end
+end
+
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `CustomField)
+
+  let read id =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/custom_field/entity_guard.ml]: used by cqrs commands now

File: pool/app/i18n/entity_guard.ml [hunk 8]
```diff
@@ -2,13 +2,24 @@ module Target = struct
...
+  ;;
+end
+
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `I18n)
+
+  let read id =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/i18n/entity_guard.ml]: used in web handler

File: pool/app/message_template/entity_guard.ml [hunk 9]
```diff
@@ -2,13 +2,26 @@ module Target = struct
...
-    |> Lwt_result.map_error Pool_common.Message.authorization
+    >|- Pool_common.Message.authorization
   ;;
 end
+
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let read id =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/message_template/entity_guard.ml]: removed

File: pool/app/pool_tenant/entity_guard.ml [hunk 10]
```diff
@@ -1,47 +1,67 @@
...
+  ;;
+end
+
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `Tenant)
+
+  let read id =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/pool_tenant/entity_guard.ml]: is used in web handler

File: pool/queue/entity_guard.ml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+module Access = struct
+  open Guard
+  open ValidationSet
+
+  let index = One (Action.Read, TargetSpec.Entity `Queue)
+  let read = One (Action.Read, TargetSpec.Entity `Queue)
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/queue/entity_guard.ml:6]: used for detail view of queue elements

File: pool/routes/routes.ml (hunk 2 above)
Comment (mabiede) [reply pool/routes/routes.ml:558]: refactored

File: pool/routes/routes.ml (hunk 1 above)
Comment (mabiede) [reply pool/routes/routes.ml:388]: refactored

**Pull request merged**